### PR TITLE
fix: convert antd to typescript

### LINKED ISSUE #3123: Wrong import in rjsf/antd
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/)
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

antd

### Version

5.0.0-beta.8

### Current Behavior

Wrong react import in generated file
node_modules/@rjsf/antd/dist/index.d.ts:35
```typescript
export const Form: import("@rjsf/core/node_modules/@types/react").ForwardRefExoticComponent<import("@rjsf/core").FormProps<any, {}> & import("@rjsf/core/node_modules/@types/react").RefAttributes<import("@rjsf/core").default<any, {}>>>;
```

### Expected Behavior

Must be `import("@types/react")`

### Steps To Reproduce

_No response_

### Environment

```markdown
- OS: macOS 12.5
- Node: v18.7.0
- npm: 8.15.0
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by heath-freenome**
### Reasons for making this change

The conversion to Typescript fixes #3123

- Updated `.eslintrc` to use the typescript version
- Updated `jest.config.js` to incorporate the `jest` block refactored from `package.json`
- Update `package.json` to switch from `js` to `ts?(x)`, adding necessary type imports and `typings` block
- Refactored all of the `.js` file to Typescript
  - Any files that used the `withConfigConsumer()` HOC needed to define a custom type that included `prefixCls: string`
  - Had to do some `extraProps` spreading hacks for those component types that accepted props, but didn't have them defined properly on the type
  - Added an `index.ts` for the `templates` and `widgets` directories, simpifying the main `index.ts`
- Converted the tests, renaming the snapshots accordingly, updating those templates that differed by just some whitespace rendering

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/antd/src/templates/DescriptionField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { DescriptionFieldProps } from "@rjsf/utils";
+
+const DescriptionField = ({
+  description,
+  id,
+}: // registry,
```
Comment (nickgros) [inline packages/antd/src/templates/DescriptionField/index.tsx:7]:
```suggestion
}:
```

File: packages/antd/src/templates/ErrorList/index.tsx [hunk 2]
```diff
@@ -7,11 +8,11 @@ import ExclamationCircleOutlined from "@ant-design/icons/ExclamationCircleOutlin
 
 const ErrorList = ({
   // errorSchema,
```
Comment (nickgros) [inline packages/antd/src/templates/ErrorList/index.tsx:10]:
```suggestion
```

File: packages/antd/src/templates/ErrorList/index.tsx [hunk 3]
```diff
@@ -7,11 +8,11 @@ import ExclamationCircleOutlined from "@ant-design/icons/ExclamationCircleOutlin
...
-  errors,
-  // formContext,
   // schema,
   // uiSchema,
```
Comment (nickgros) [inline packages/antd/src/templates/ErrorList/index.tsx:12]:
```suggestion
```

File: packages/antd/src/templates/ErrorList/index.tsx [hunk 4]
```diff
@@ -7,11 +8,11 @@ import ExclamationCircleOutlined from "@ant-design/icons/ExclamationCircleOutlin
...
-}) => {
+  // formContext,
```
Comment (nickgros) [inline packages/antd/src/templates/ErrorList/index.tsx:13]:
```suggestion
```

File: packages/antd/src/widgets/CheckboxWidget/index.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  id,
+  label,
+  onBlur,
+  onChange,
+  onFocus,
+  // options,
+  // placeholder,
+  readonly,
+  // required,
+  // schema,
```
Comment (nickgros) [inline packages/antd/src/widgets/CheckboxWidget/index.tsx:18]:
```suggestion
  readonly,
```

File: packages/antd/src/widgets/CheckboxWidget/index.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+  const handleFocus = ({ target }: React.FocusEvent<HTMLInputElement>) =>
+    onFocus(id, target.checked);
+
+  // Antd's typescript definitions do not contain the following props that are actually necessary and, if provided,
+  // they are used, so hacking them in via by spreading `extraProps` on the component to avoid typescript errors
+  const extraProps = {
+    onBlur: !readonly ? handleBlur : undefined,
+    onFocus: !readonly ? handleFocus : undefined,
+  };
```
Comment (nickgros) [inline packages/antd/src/widgets/CheckboxWidget/index.tsx:37]: The types don't support them but the props still work? That's annoying!

File: packages/antd/src/widgets/DateWidget/index.tsx [hunk 7]
```diff
@@ -22,17 +23,17 @@ const DateWidget = ({
   // required,
   // schema,
```
Comment (nickgros) [inline packages/antd/src/widgets/DateWidget/index.tsx:24]:
```suggestion
```

File: packages/antd/src/widgets/PasswordWidget/index.tsx [hunk 8]
```diff
@@ -17,17 +17,19 @@ const PasswordWidget = ({
   // required,
   // schema,
```
Comment (nickgros) [inline packages/antd/src/widgets/PasswordWidget/index.tsx:18]:
```suggestion
```

File: packages/antd/src/widgets/RadioWidget/index.tsx [hunk 9]
```diff
@@ -18,25 +17,27 @@ const RadioWidget = ({
   // required,
```
Comment (nickgros) [inline packages/antd/src/widgets/RadioWidget/index.tsx:17]:
```suggestion
```

File: packages/antd/src/widgets/RangeWidget/index.tsx [hunk 10]
```diff
@@ -19,35 +17,40 @@ const RangeWidget = ({
   // required,
```
Comment (nickgros) [inline packages/antd/src/widgets/RangeWidget/index.tsx:17]:
```suggestion
```

**Review by nickgros — APPROVED** [review]
Have suggested changes but they don't block approval. Thanks @heath-freenome !

File: packages/antd/src/templates/DescriptionField/index.tsx (hunk 1 above)
Comment (heath-freenome) [reply packages/antd/src/templates/DescriptionField/index.tsx:7]: Sounds like a great plan!

File: packages/antd/src/widgets/CheckboxWidget/index.tsx (hunk 6 above)
Comment (heath-freenome) [reply packages/antd/src/widgets/CheckboxWidget/index.tsx]: Done, although, oddly, once I add it for one prop, the other props stop complaining

Comment (heath-freenome) [reply packages/antd/src/widgets/CheckboxWidget/index.tsx]: Hmmm, ok, that didn't work so I'm reverting back to the extraProps... see https://github.com/rjsf-team/react-jsonschema-form/actions/runs/3129403515/jobs/5078449932

**Pull request merged**